numpy/core/tests/test_getlimits.py

```diff
@@ -48,12 +48,19 @@ def test_basic(self):
         for dt1, dt2 in dts:
             for attr in ('bits', 'eps', 'epsneg', 'iexp', 'machep',
                          'max', 'maxexp', 'min', 'minexp', 'negep', 'nexp',
-                         'nmant', 'precision', 'resolution', 'tiny',
-                         'smallest_normal', 'smallest_subnormal'):
+                         'nmant', 'mant_dig', 'precision', 'resolution',
+                         'tiny', 'smallest_normal', 'smallest_subnormal'):
                 assert_equal(getattr(finfo(dt1), attr),
                              getattr(finfo(dt2), attr), attr)
         assert_raises(ValueError, finfo, 'i4')
 
+    def test_mant_dig(self):
+        dts = list(zip(['f2', 'f4', 'f8', 'c8', 'c16'],
+                       [np.float16, np.float32, np.float64, np.complex64,
+                        np.complex128]))
+        for dt in dts:
+            assert_equal(finfo(dt).nmant + 1, finfo(dt).mant_dig)
```

Comment:
Do we have good tests for `nmant`?  I might suggest adding a test using
```
one = dt.type(1)
next = np.nextafter(one, dt.type(2))
diff = next - one
nmant_expected = 1 - np.log2(diff)
```
Although, I suppose for `double double` this needs some care.  OTOH, I am not actually sure yet whether we should report 106 or 107 there!

Reply:
>one = dt.type(1)

I try to run that in the command line, but it doesn't work.  I think the tests are good enough as they are, since we're really just checking to see if mant_dig is 1 larger than nmant.

Reply:
IMO it'd be better to formulate the tests so that the result does not reference `nmant`, since it's stated that `nmant` may eventually be removed.

Reply:
clearly, `dt` here is the same `dt` as is already in the test!

Reply:
> clearly, `dt` here is the same `dt` as is already in the test!

I know. When I said that I it didn't run in the command line, I meant that I checked to see if it would work in the command line:
`
        dts = list(zip(['f2', 'f4', 'f8', 'c8', 'c16'],
                       [np.float16, np.float32, np.float64, np.complex64,
                        np.complex128]))
        for dt in dts:
            print(dt.type(1))
`
I probably could have explained that better, tbh.